app/services/saml_request_validator.rb

```diff
@@ -113,17 +113,17 @@ def ial_max_requested?
     Array(authn_context).include?(Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF)
   end
 
-  def authorized_email_nameid_format
-    return unless email_nameid_format?
-    return if service_provider&.email_nameid_format_allowed
+  def authorized_nameid_format
+    return if nameid_format.nil?
+    return if email_nameid_format? && service_provider&.email_nameid_format_allowed
```

Comment:
🤔 i would maybe pull out these multi-conditionals and turn them into well-named method that explain a little more clearly what the condition is, but i don't think that's blocking.

Reply:
I was thinking along the same lines but stopped short. Thank you for the nudge.